lib/Controller/RecordingController.php

```diff
@@ -26,32 +26,96 @@
 namespace OCA\Talk\Controller;
 
 use InvalidArgumentException;
+use GuzzleHttp\Exception\ConnectException;
 use OCA\Talk\Config;
-use OCA\Talk\Exceptions\UnauthorizedException;
 use OCA\Talk\Service\RecordingService;
-use OCA\Talk\Service\SIPBridgeService;
 use OCP\AppFramework\Http;
 use OCP\AppFramework\Http\DataResponse;
+use OCP\Http\Client\IClientService;
 use OCP\IRequest;
+use Psr\Log\LoggerInterface;
 
 class RecordingController extends AEnvironmentAwareController {
 	public function __construct(
 		string $appName,
 		IRequest $request,
+		private ?string $userId,
 		private Config $talkConfig,
-		private SIPBridgeService $SIPBridgeService,
-		private RecordingService $recordingService
+		private IClientService $clientService,
+		private RecordingService $recordingService,
+		private LoggerInterface $logger
 	) {
 		parent::__construct($appName, $request);
 	}
 
+	public function getWelcomeMessage(int $serverId): DataResponse {
```

Comment:
Is this only used for the admin page? Or also by the recording server?

Reply:
Only for the admin page (it is the sibling of [`SignalingController::getWelcomeMessage()`](https://github.com/nextcloud/spreed/blob/5f060687c8a0c77f1a61bf9a3462d4bfa4fe055f/lib/Controller/SignalingController.php#L197).